packages/playwright-test/src/index.ts

```diff
@@ -109,20 +104,31 @@ const playwrightFixtures: Fixtures<TestFixtures, WorkerFixtures> = ({
       options.channel = channel;
 
     for (const browserType of [playwright.chromium, playwright.firefox, playwright.webkit]) {
-      (browserType as any)._defaultLaunchOptions = options;
-      (browserType as any)._defaultConnectOptions = connectOptions;
+      (browserType as BrowserTypeImpl)._defaultLaunchOptions = options;
+      (browserType as BrowserTypeImpl)._defaultConnectOptions = connectOptions;
     }
     await use(options);
     for (const browserType of [playwright.chromium, playwright.firefox, playwright.webkit]) {
-      (browserType as any)._defaultLaunchOptions = undefined;
-      (browserType as any)._defaultConnectOptions = undefined;
+      (browserType as BrowserTypeImpl)._defaultLaunchOptions = undefined;
+      (browserType as BrowserTypeImpl)._defaultConnectOptions = undefined;
     }
   }, { scope: 'worker', auto: true }],
 
-  browser: [async ({ playwright, browserName }, use, testInfo) => {
+  browser: [async ({ playwright, browserName, connectOptions }, use, testInfo) => {
     if (!['chromium', 'firefox', 'webkit'].includes(browserName))
       throw new Error(`Unexpected browserName "${browserName}", must be one of "chromium", "firefox" or "webkit"`);
-    const browser = await playwright[browserName].launch();
+    let browser: Browser;
+    if (connectOptions && process.env.PW_TEST_REUSE_CONTEXT) {
+      browser = await (playwright[browserName] as BrowserTypeImpl)._connectInsteadOfLaunching({
+        ...connectOptions,
+        headers: {
+          ...connectOptions.headers,
+          'x-playwright-reuse-context': '1',
+        },
+      }, {});
+    } else {
+      browser = await playwright[browserName].launch();
```

Comment:
I'm not sure I follow what is happening here since launch also goes into connectInsteadOfLaunching when we have connect options. Looks like you are building a mode where only fixture's browser is using the connection. If seems like not setting `_defaultConnectOptions` in that case would achieve that.